MGDAPI-3374 update bundle rhmi operators script and readme

MStokluska opened the pull request:
# Issue link
https://issues.redhat.com/browse/MGDAPI-3374

# What
- update readme
- update bundle rhmi operator to support bundle format

# Verification steps
From this branch run:
```
INSTALLATION_TYPE=managed-api make cluster/preapare/local
```
```
INSTALLATION_TYPE=managed-api make deploy/integreatly-rhmi-cr.yml
```
```
make create/olm/bundle OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.16.0,1.15.2 ORG=<YOUR_QUAY_ORG> OC_INSTALL=true`
```
- And confirm that a catalog source for 1.15.2 has been created on your cluster, navigate to operators hub, find RHOAM, install under redhat-rhoam-operator ns. 
- Once installation completes, navigate to same catalog source that 1.15.2 was referred from, update it to 1.16.0.
- Navigate to installed operators and click approve on RHOAM upgrade. 
- Confirm that the upgrade has been successful.

austincunningham commented on scripts/bundle-rhmi-operators.sh:
```diff
@@ -126,18 +130,10 @@ push_index() {
 }
 
-# creates catalog source on the cluster
-create_catalog_source() {
```
why remove this , is it no longer valid ?

MStokluska replied:
It actually  only works as expected with a single image and gets too messy for what's its giving when it comes to more index images

MStokluska replied:
Actually found a reasonable solution. 
Added it back in but only the latest version is picked up, also, only if it has no replaces field. This seems to me like only reasonable solution. Let me know if you have any other ideas

MStokluska commented on scripts/bundle-rhmi-operators.sh:
```diff
@@ -1,15 +1,31 @@
+# - opm
+# - operator-sdk
+# Function:
+# Builds and pushes the bundles and indexes for RHMI or/and RHOAM
```
Sure, added indexImage list

MStokluska replied:
> 

@austincunningham this feels to me like there was something else done wrong. 
I've updated the verification part, can you double check if your repos on quay are all public?

MStokluska commented:
> being trying to verify this today of master but the operator fails to install
> 
>     * copied and replace the script https://github.com/integr8ly/integreatly-operator/blob/57ad7bbe35f14eefaa26a816b3e686968db5b5e6/scripts/bundle-rhmi-operators.sh
> 
>     * copied and replaced the bundle.Dockerfile https://github.com/integr8ly/integreatly-operator/blob/57ad7bbe35f14eefaa26a816b3e686968db5b5e6/bundles/managed-api-service/bundle.Dockerfile
> 
>     * add back in the `make create/olm/bundle` make command
> 
>     * commands run
> 
> 
> ```shell
> export ORG=austincunningham
> OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/build
> OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/push
> OLM_TYPE=managed-api-service SEMVER=1.16.0 make release/prepare
> make create/olm/bundle OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.16.0 ORG=austincunningham
> # update the index image in the catalogsource 
> oc apply -f catalogSource.yaml
> ```
> 
> operator hub install times out for RHOAM ![image](https://user-images.githubusercontent.com/16667688/152380551-082bd457-b060-4f66-b9a7-2b3357aef54f.png)
@austincunningham this feels to me like there was something else done wrong.
I've updated the verification part, can you double check if your repos on quay are all public?

MStokluska commented:
> > being trying to verify this today of master but the operator fails to install
> > ```
> > * copied and replace the script https://github.com/integr8ly/integreatly-operator/blob/57ad7bbe35f14eefaa26a816b3e686968db5b5e6/scripts/bundle-rhmi-operators.sh
> > 
> > * copied and replaced the bundle.Dockerfile https://github.com/integr8ly/integreatly-operator/blob/57ad7bbe35f14eefaa26a816b3e686968db5b5e6/bundles/managed-api-service/bundle.Dockerfile
> > 
> > * add back in the `make create/olm/bundle` make command
> > 
> > * commands run
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```shell
> > export ORG=austincunningham
> > OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/build
> > OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/push
> > OLM_TYPE=managed-api-service SEMVER=1.16.0 make release/prepare
> > make create/olm/bundle OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.16.0 ORG=austincunningham
> > # update the index image in the catalogsource 
> > oc apply -f catalogSource.yaml
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > operator hub install times out for RHOAM ![image](https://user-images.githubusercontent.com/16667688/152380551-082bd457-b060-4f66-b9a7-2b3357aef54f.png)
> > @austincunningham this feels to me like there was something else done wrong.
> > I've updated the verification part, can you double check if your repos on quay are all public?

@austincunningham I think I know what's wrong, 
1.16.0 bundles was referencing an rc-1 image. This is corrected now as part of this PR to reference the actual image used in stage, but, when you run `make realase/prepare` for 1.16.0, you would update the image to `containerImage: quay.io/integreatly/managed-api-service:rhoam-v1.16.0` in the CSV. 
Because of all the changes around our releases, this image is no longer present. 
So my guess here is when you did the make image/build and image/push you have not updated the CSV with the image you have just pushed. If you did then I'm out of ideas and I think we might debug this together. 
My installation of RHOAM using the bundle script works fine: (but had to update the metadata of 1.15.2 and 1.15.0)
![image](https://user-images.githubusercontent.com/43473912/152407472-c597396f-b4e8-4269-a98f-7c926bdd78da.png)

laurafitzgerald commented on scripts/bundle-rhmi-operators.sh:
```diff
@@ -124,20 +140,24 @@ push_index() {
   printf 'Pushing index image:'$INDEX_IMAGE'\n'
 
   docker push $INDEX_IMAGE
```
```suggestion
  $BUILD_TOOL push $INDEX_IMAGE
```

laurafitzgerald commented on scripts/bundle-rhmi-operators.sh:
```diff
@@ -79,15 +98,12 @@ generate_bundles() {
+    pwd
+    cd ../../..
+    docker build -f ./bundles/$OLM_TYPE/bundle.Dockerfile -t $REG/$ORG/${OLM_TYPE}-bundle:$VERSION --build-arg manifest_path=./bundles/$OLM_TYPE/temp/$VERSION/manifests --build-arg metadata_path=./bundles/$OLM_TYPE/temp/$VERSION/metadata --build-arg version=$VERSION .
     docker push $REG/$ORG/${OLM_TYPE}-bundle:$VERSION
```
```suggestion
    $BUILD_TOOL push $REG/$ORG/${OLM_TYPE}-bundle:$VERSION
```

laurafitzgerald commented on scripts/bundle-rhmi-operators.sh:
```diff
@@ -79,15 +98,12 @@ generate_bundles() {
-    docker build -f bundle.Dockerfile -t $REG/$ORG/${OLM_TYPE}-bundle:$VERSION .
+    pwd
+    cd ../../..
+    docker build -f ./bundles/$OLM_TYPE/bundle.Dockerfile -t $REG/$ORG/${OLM_TYPE}-bundle:$VERSION --build-arg manifest_path=./bundles/$OLM_TYPE/temp/$VERSION/manifests --build-arg metadata_path=./bundles/$OLM_TYPE/temp/$VERSION/metadata --build-arg version=$VERSION .
```
```suggestion
    $BUILD_TOOL build -f ./bundles/$OLM_TYPE/bundle.Dockerfile -t $REG/$ORG/${OLM_TYPE}-bundle:$VERSION --build-arg manifest_path=./bundles/$OLM_TYPE/temp/$VERSION/manifests --build-arg metadata_path=./bundles/$OLM_TYPE/temp/$VERSION/metadata --build-arg version=$VERSION .
```

austincunningham commented:
# Verified
- I have verified these changes building development images
```bash
export ORG=austincunningham
OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/build
OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api make image/push
OLM_TYPE=managed-api-service SEMVER=1.16.0 make release/prepare
# after running release/prepare you need to edit the csv in the bundles dir to remove the rhoam- part of the tag and the replaces field
# you can then run the following command to create the bundles and index
make create/olm/bundle OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.16.0 
# update the index image in the catalogsource 
oc apply -f catalogSource.yaml 
```
deployed as expected
/lgtm
I do think the `$BUILD_TOOL` suggestion is worth adding

austincunningham commented:
/lgtm

MStokluska commented:
/approve
